core/init/db/db.go

```diff
@@ -1,89 +1,13 @@
 package db
 
 import (
-	"fmt"
-	"log"
-	"os"
 	"path"
-	"time"
 
 	"github.com/1Panel-dev/1Panel/core/global"
-	"github.com/glebarez/sqlite"
-	"gorm.io/gorm"
-	"gorm.io/gorm/logger"
+	"github.com/1Panel-dev/1Panel/core/utils/common"
 )
 
 func Init() {
-	initDB()
-	initTaskDB()
-}
-
-func initDB() {
-	dbPath := path.Join(global.CONF.System.BaseDir, "1panel/db")
-	if _, err := os.Stat(dbPath); err != nil {
-		if err := os.MkdirAll(dbPath, os.ModePerm); err != nil {
-			panic(fmt.Errorf("init db dir failed, err: %v", err))
-		}
-	}
-	fullPath := path.Join(dbPath, global.CONF.System.DbCoreFile)
-	if _, err := os.Stat(fullPath); err != nil {
-		f, err := os.Create(fullPath)
-		if err != nil {
-			panic(fmt.Errorf("init db file failed, err: %v", err))
-		}
-		_ = f.Close()
-	}
-
-	db, err := NewDBWithPath(fullPath)
-	if err != nil {
-		panic(err)
-	}
-
-	global.DB = db
-	global.LOG.Info("init db successfully")
-}
-
-func initTaskDB() {
-	fullPath := path.Join(global.CONF.System.BaseDir, "1panel/db/task.db")
-	if _, err := os.Stat(fullPath); err != nil {
-		f, err := os.Create(fullPath)
-		if err != nil {
-			panic(fmt.Errorf("init task db file failed, err: %v", err))
-		}
-		_ = f.Close()
-	}
-
-	db, err := NewDBWithPath(fullPath)
-	if err != nil {
-		panic(err)
-	}
-
-	global.TaskDB = db
-	global.LOG.Info("init task db successfully")
-}
-
-func NewDBWithPath(dbPath string) (*gorm.DB, error) {
-	db, _ := gorm.Open(sqlite.Open(dbPath), &gorm.Config{
-		DisableForeignKeyConstraintWhenMigrating: true,
-		Logger:                                   getLogger(),
-	})
-	sqlDB, dbError := db.DB()
-	if dbError != nil {
-		return nil, dbError
-	}
-	sqlDB.SetConnMaxIdleTime(10)
-	sqlDB.SetMaxOpenConns(100)
-	sqlDB.SetConnMaxLifetime(time.Hour)
-	return db, nil
-}
-func getLogger() logger.Interface {
-	return logger.New(
-		log.New(os.Stdout, "\r\n", log.LstdFlags),
-		logger.Config{
-			SlowThreshold:             time.Second,
-			LogLevel:                  logger.Silent,
-			IgnoreRecordNotFoundError: true,
-			Colorful:                  false,
-		},
-	)
+	global.DB = common.LoadDBConnByPath(path.Join(global.CONF.System.BaseDir, "1panel/db/core.db"), "core")
+	global.TaskDB = common.LoadDBConnByPath(path.Join(global.CONF.System.BaseDir, "1panel/db/task.db"), "task")
 }
```

Comment:
This is a piece of Go code that seems to setup a database connection based on the provided configuration settings. There's not much specific information about errors, issues or optimizations from my perspective since there isn't any detailed debugging or logging available within this code snippet.

I'd recommend using additional debug statements in future versions if you need more details like the types and contents of passed parameters, error messages, log output etc., which aren't visible here due to lack of context. This can help catch bugs quicker during development. Additionally, it would be good when these operations happen only once at startup rather than on every request, which might lead them to become slow over time, especially with very big configurations or larger volumes of data.

Remember though, unless stated otherwise, I'm working under the assumption that all required packages are imported correctly. If an import statement doesn’t work for you (for example `import { ... }` in JavaScript), then please include where it does work.